src/archivematicaCommon/lib/externals/checksummingTools.py

```diff
@@ -1,26 +1,37 @@
 #!/usr/bin/env python2
+
+from __future__ import print_function
+
 import hashlib
+import sys
 
-#Borrowed from http://stackoverflow.com/questions/1131220/get-md5-hash-of-a-files-without-open-it-in-python
-def md5_for_file(filename):
-    md5 = hashlib.md5()
-    with open(filename, 'rb') as f:
-        for chunk in iter(lambda: f.read(2**10 * md5.block_size), b''):
-            md5.update(chunk)
-    return md5.hexdigest()
 
+def get_file_checksum(filename, algorithm='sha256'):
+    """
+    Perform a checksum on the specified file.
+
+    This function reads in files incrementally to avoid memory exhaustion.
+    See: http://stackoverflow.com/questions/1131220/get-md5-hash-of-a-files-without-open-it-in-python
+
+    :param filename: The path to the file we want to check
+    :param algorithm: Which algorithm to use for hashing, e.g. 'md5'
+    :return: Returns a checksum string for the specified file.
+    """
+    h = hashlib.new(algorithm)
```

Comment:
Sure, I can check `if algorithm and algorithm in hashlib.algorithms` or something like that. Do you think we should instantiate a logger here to log the error or just throw an exception?